networkx/algorithms/bipartite/matrix.py

```diff
@@ -112,26 +112,61 @@ def biadjacency_matrix(
 
 
 @nx._dispatchable(graphs=None, returns_graph=True)
-def from_biadjacency_matrix(A, create_using=None, edge_attribute="weight"):
+def from_biadjacency_matrix(
+    A,
+    create_using=None,
+    edge_attribute="weight",
+    top_nodelist=None,
+    bottom_nodelist=None,
```

Comment:
I think we should look more closely at established convention for the top/bottom arguments. It looks like `top_nodes=` is frequently used, and `bottom_nodes` is implied (so it's not a keyword argument).

Also, when adding new keyword arguments, we sometimes consider making them keyword only by using e.g. `*, top_nodes=None` in the signature. @rossbar, do you think keyword-only makes sense here?

Reply:
Good suggestions -- at least from my perspective:
- only input top_nodes
- make that new input keyword-only

I wonder if we have in the contributing docs that we prefer keyword only for new parameters. We should have that somewhere. :)

Reply:
> I think we should look more closely at established convention for the top/bottom arguments. It looks like top_nodes= is frequently used, and bottom_nodes is implied (so it's not a keyword argument).

I think this is generally true, though IIUC the goal here is to provide a label for the nodes (i.e. instead of integers) in which case I wonder if the better approach isn't to break from the `nodelist` pattern and instead use an explicit mapping (though this assumes that the user will know what the integer indices will be a priori)

Reply:
Ah, crap. Did not refresh my page recently enough to see @rossbar's comment. 

> though IIUC the goal here is to provide a label for the nodes (i.e. instead of integers)

Yep, this is the goal here.

> in which case I wonder if the better approach isn't to break from the nodelist pattern and instead use an explicit mapping (though this assumes that the user will know what the integer indices will be a priori)

This sounds like it could be a good idea to me. In any implementation, the user will need to know the indices a priori anyhow to meaningfully construct the `nodelists` with correct ordering. I suppose mapping could add additional confusion due to taking the integer of the node for the first row/column to 1 rather than 0, leading to an off-by-one error.

Reply:
All good points @LorentzFactor. Thank you!
We are indicating more than which node is in which set -- we are indicating the order of the nodes in each set.
So indeed -- we need two input parameters and the phrase `nodelist` in the names makes good sense too.